packages/terra-application/src/primary-navigation-layout/workspace-layout/PrimaryNavigationWorkspace.jsx

```diff
@@ -0,0 +1,147 @@
+   */
+  onSizeChange: PropTypes.func,
+  /**
+   * Callback function triggering when the presentation state changes.
+   */
+  onPresentationStateChange: PropTypes.func,
+  /**
+   * Callback function triggering when the active.
+   */
+  onActiveItemChange: PropTypes.func,
+  /**
+   * The string key linked to the action workspace.
+   */
+  initialActiveItemKey: PropTypes.string.isRequired,
+  /**
+   * Child content to be placed within the main content region.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Whether or not the workspace should initially display as open.
+   */
+  initialIsOpen: PropTypes.bool,
+  /**
+   * The size string value matching the active size option.
+   */
+  initialSize: PropTypes.object,
+  /**
+   * @private
+   * Id string to apply to the workspace
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * @private
+   * Whether or not the workspace is open
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * @private
+   * Function callback i.e. `onRequest(event)`
+   */
+  onRequestClose: PropTypes.func,
+  /**
+   * @private
+   * Whether or not the workspace is present as an overlay
+   */
+  isPresentedAsOverlay: PropTypes.bool,
+  /**
+   * @private
+   * Numeric scale value ranging from `0.0 - 1.0` as the minimum to maximum size for the workspace
+   */
+  sizeScalar: PropTypes.number,
+  /**
+   * @private
+   * The string representation of the workspace size
+   */
+  activeSize: PropTypes.string,
+  /**
+   * @private
+   * Array of objects containing key/text pairs for the available size options
+   */
+  sizeOptions: PropTypes.arrayOf(PropTypes.shape({
+    key: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+  })),
+  /**
+   * @private
+   * Function callback i.e. `onRequestSizeChange(size)`
+   */
+  onRequestSizeChange: PropTypes.func,
+};
+
+const WorkspaceWrapper = ({
+  // consumer props
+  onSizeChange,
+  onPresentationStateChange,
+  onActiveItemChange,
+  initialActiveItemKey,
+  children,
+  initialIsOpen, // eslint-disable-line no-unused-vars, react/no-unused-prop-types
+  initialSize, // eslint-disable-line no-unused-vars, react/no-unused-prop-types
+
+  // private injected props
+  id,
+  isOpen,
+  onRequestClose,
+  isPresentedAsOverlay,
+  sizeScalar,
+  activeSize,
+  sizeOptions,
+  onRequestSizeChange,
+}) => {
+  const [activeItemKey, setActiveItemKey] = React.useState(initialActiveItemKey);
+
+  React.useEffect(() => {
+    if (onSizeChange && sizeScalar !== undefined) {
+      onSizeChange(sizeScalar);
+    }
+  }, [sizeScalar, onSizeChange]);
+
+  React.useEffect(() => {
+    if (onPresentationStateChange) {
+      onPresentationStateChange(isOpen);
+    }
+  }, [isOpen, onPresentationStateChange]);
+
+  React.useEffect(() => {
+    if (onActiveItemChange) {
+      onActiveItemChange(activeItemKey);
+    }
+  }, [activeItemKey, onActiveItemChange]);
+
+  return (
+    <Workspace
+      id={id || 'test-id'}
```

Comment:
is this the fallback id that we want?

Reply:
This might be unnecessary now, I'll validate and remove.